tests/Feature/FlightController/IndexTest.php

```diff
@@ -0,0 +1,172 @@
+                    'id' => 'integer',
+                    'flight_number' => 'string',
+                    'origin_city_id' => 'integer',
+                    'destination_city_id' => 'integer',
+                    'departure_at' => 'string',
+                    'arrival_at' => 'string',
+                    'created_at' => 'string',
+                    'updated_at' => 'string'
+                ])
+                       ->etc()
+            )
+                ->etc()
+        );
+    }
+
+    public function test_flights_have_origin_city_relationship(): void
+    {
+        $response = $this->get(self::ENDPOINT);
+
+        $response->assertJson(fn (AssertableJson $json) =>
+            $json->has('data.0', fn(AssertableJson $flight) =>
+                $flight->where('origin', $this->flight->origin)
+                       ->etc()
+            )
+                 ->etc()
+        );
+    }
+
+    public function test_flights_have_destination_city_relationship(): void
+    {
+        $response = $this->get(self::ENDPOINT);
+
+        $response->assertJson(fn (AssertableJson $json) =>
+            $json->has('data.0', fn(AssertableJson $flight) =>
+                $flight->where('destination', $this->flight->destination)
+                       ->etc()
+            )
+                 ->etc()
+        );
+    }
+
+    public function test_flights_have_airline_relationship(): void
+    {
+        $response = $this->get(self::ENDPOINT);
+
+        $response->assertJson(fn (AssertableJson $json) =>
+            $json->has('data.0', fn(AssertableJson $flight) =>
+                $flight->where('airline', $this->flight->airline)
+                       ->etc()
+            )
+                 ->etc()
+        );
+    }
+
+    public function test_response_returns_10_flights_even_if_more_on_the_db(): void
+    {
+        $this->seed();
+
+        $response = $this->get(self::ENDPOINT);
+
+        $response->assertJson(fn (AssertableJson $json) =>
+            $json->has('data', 10)
+                 ->etc()
+        );
+
+        $this->assertDatabaseCount('flights', 51);
+    }
+
+    public function test_response_has_pagination_links(): void
+    {
+        $this->seed();
+
+        $response = $this->get(self::ENDPOINT);
+
+        $response->assertJson(fn (AssertableJson $json) =>
+            $json->has('links')
+                 ->has('links.0', fn(AssertableJson $paginationLink) =>
+                    $paginationLink->has('url')
+                                   ->has('label')
+                                   ->has('active')
+                 )
+                 ->etc()
+        );
+    }
+
+    public function test_pagination_links_mantain_query_string(): void
+    {
+        $this->seed();
+
+        $endpoint = self::ENDPOINT . '?hello=world';
+
+        $response = $this->get($endpoint);
+
+        $response->assertJsonPath(
+            'links.2.url',
+            'http://localhost/' . $endpoint . '&page=2'
```

Comment:
not really a good idea to put urldomain in tests for this case, the test should be as agnostic as possible